platform/mds/mds-web/src/main/resources/webapp/js/controllers.js

```diff
@@ -3743,13 +3747,34 @@
         */
         $scope.addEntityInstance = function () {
             blockUI();
-
-            var values = $scope.currentRecord.fields;
+            $scope.addEntityInstanceBase($scope.currentRecord);
+        };
+        $scope.addEntityInstanceBase = function (inst)
+        {
+            var values = inst.fields;
             angular.forEach (values, function(value, key) {
                 value.value = value.value === 'null' ? null : value.value;
             });
+            for(var i in inst.fields) {
+                var fd = inst.fields[i];
+
+                // {readonly, uiFilterable, . . .} = fields(FieldDto) - fields(FieldRecord)
+                delete fd["readOnly"];
```

Comment:
FieldRecord was geared towards the data browser, while fieldDto for the schema view. The only noteworthy difference I can see is that field records holds the value for the field. I think we could refactor FieldRecord to inherit from FieldDto in the future.